### ISSUE TITLE
Content Delivery Server (Azure WebApps) redirects are not working

### ISSUE THREAD

**Issue by Ricardowouda** [iamandycohen/UrlRewrite#36]
Hi,

On our Content Delivery server the redirects are not working.

I've installed the module(v.1.8.2) on Sitecore 8.2 (rev. 161221) on a Azure Web Apps environment. On the Content Management Servers everything is working fine. The redirects are working and also the UrlRewriter module in sitecore is working fine for all databases (master and web). 

However on the Content Delivery server the redirects are not working at all. I don't get any errors in the logs. The Hi.UrlRewrite.dll and the config files are existing on the Content Delivery server. 

Any idea what could be going on and how I can fix this?

**Comment by iamandycohen** [comment]
My guess is that you need to make sure the links database is set
correctly.  This module was first designed specifically for Azure.

**Comment by Ricardowouda** [comment]
Thanks for your reply, it seems to work now a bit better. The linkdatabase configuration wasn't correctly. However on each redirect you must be logged in into Sitecore for some reason. When I go to a url which is configured as a redirect, I get the Sitecore login page (if not logged in). This also happens on the Content Delivery server where you can't login into Sitecore. So I need to be logged in into Sitecore for some reason. 

The target page is just a page which is available without Sitecore login. For example even when I try to redirect to google.com, it also redirects to the Sitecore login page.

Any idea what could be going on and how I can fix this?

**Comment by iamandycohen** [comment]
You have published all the redirects and their patent folders, correct?

**Comment by Ricardowouda** [comment]
Yes I have published them all. All items are correctly on the Web db. Also on the Content Management server when not logged in I have this issue.

**Comment by iamandycohen** [comment]
Do you have any Modules installed?  Are you able to email me the contents
of your App_Config folder and also your web.config on all your servers?
CM, Publishing, Processing and CD?

**Comment by Ricardowouda** [comment]
We don't have other modules installed yet. I'll send you an e-mail with the files. 
Thanks for your help

**Comment by HaikoF** [comment]
Just a simple idea .... 
Is your site configured correct - do you have configured for your site a loginPage? 
And are there any security settings on the redirect items or the parent folder that prevent anonymous/everyone from reading the item(s)?

**Comment by Ricardowouda** [comment]
There is no login page configured for the current site where the redirects items are configured. The items have read rights for everyone. Same rights as the homepage for example. 
See this example (I've only enabled the first redirect item Simple External Redirect): 
![access redirect items](https://user-images.githubusercontent.com/17002270/30153189-0c1b2fe4-93b6-11e7-8e17-ac5fb13e7fc0.PNG)

**Comment by mattyindustries** [comment]
I believe that we have encountered the same issue running in Azure Web Apps

The default Sitecore Web.Configs when running in Cloud Mode contain the following restriction which only allows access to authenticated users

`
  <location path="sitecore modules/Shell">
    <system.web>
      <authorization>
        <deny users="?" />
        <allow users="*" />
      </authorization>
    </system.web>
  </location>
`

And the UrlRewriter.aspx is in this location.

We are planning on working around this issue by moving UrlRewriter.aspx into layouts\UrlRewrite\UrlRewriter.aspx but would be good to know if there is upstream plans to workaround this?

**Comment by sesebon** [comment]
If I may chime in, I'd like to ask if there's any particular instruction to install this module on the CD servers?

**Comment by davvves** [comment]
@mattyindustries I've confirmed that the workaround you propose above makes this module work with Sitecore 9. The web.configs included in Sitecore 9 by default also contain that section. I'm guessing it'll have to be a change going forward.

**Comment by mohdkhanarm** [comment]
We have upgraded our website from sitecore 8.2 to Sitecore 9.0.1. Installed Url Rewrite 8.2 and delete the sample redirect and create own redirects. everything is working fine.
We have deployed on azure Paas webapps. on CD sever manually copy dll and configs.
